vrs/marten-construction-script.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Daniel Marten 01-04-2022
+# Constructing a ~100k Variant Test Database for GA4GH-VRS Annotation Script
+# NOTE: adapted from a Jupyter Notebook, so excuse any formatting quirks! 
+
+import hail as hl
+from random import sample
+
+# importing the gnomAD v3.1.2 full variant table, approx 750,000,000 variants 
+
+ht_whole = hl.read_table('gs://gcp-public-data--gnomad/release/3.1.2/ht/genomes/gnomad.genomes.v3.1.2.sites.ht',_n_partitions=10000)
+
+
+# PREVIOUSLY CALCULATED FREQUENCY STATISTICS: 
+# frequency of the following in the general v3 database , based off of frequencties in random 10k and 100k databases
```

Comment:
Yeah I can explain this! So my goal for this project was to have 100k samples - 50k random samples, 10k general indels, 10k long reference alleles (ht.alleles[0] > 3), 10k long variant alleles (ht.alleles[1] > 3) , 10k additional sex chromosome alleles, and then 10k extra multiallelic variants. 

In a test dataset of 100k randomly sampled variants in a separate notebook, I found the fractions of indels (non-SNVs), variants with long reference alleles, variants with long variant alleles, X chromosome variants, and Y chromosome variants in there, wrote them down my physical notebook, and saved those fractions as the '_freq' variables below in the code. Those fractions were then used to obtain the desired number of variants after the hail table had been filtered. Does that make sense? Sorry if I'm not answering your question.  

I did it in a separate notebook because it's it's a bit time intensive, but sorry that's not the most transparent. Would you like to see that work here?

Reply:
thanks, this makes sense! and thanks for adding the code. quick question, though: the code you included says 1 million variants. did you use those frequencies for the sampling, or other frequencies calculated when sampling 100k variants?